lib/f_service/result/failure.rb

```diff
@@ -59,6 +59,31 @@ def value!
         raise Result::Error, 'Failure objects do not have value'
       end
 
+      # Returns its value to the given block.
+      # Use this to chain multiple service calls (since all services return Results).
+      #
+      #
+      # @example
+      #   class UsersController < BaseController
+      #     def create
+      #       result = User::Create.(user_params)
+      #                            .then { |user| User::SendWelcomeEmail.(user: user) }
+      #                            .catch { |error| Logger::Notificate.(error: error) }
```

Comment:
I think this example may lead users to a bad usage. Since `catch` will evaluate the block, this chain will return a `Success` because of `Logger::Notificate`. In this case, `on_failure` would be more adequate.

Reply:
yeah, that was a bad example indeed